### ISSUE TITLE
False positive SC2029 in connection string when using array expansion

### ISSUE THREAD

**Issue by dms-cat** [koalaman/shellcheck#1870]
#### For bugs
- Rule Id: SC2029
- My shellcheck version: 0.4.6 and web version
- [x] The rule's wiki page does not already cover this
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

options=('-N' '-T')
hostname=example.com
ssh "${options[@]}" "user@${hostname}"
```

#### Here's what shellcheck currently says:

```
In test.bash line 5:
ssh "${options[@]}" "user@${hostname}"
                          ^-- SC2029: Note that, unescaped, this expands on the client side.
```

#### Here's what I wanted or expected to see:

No output.
